packages/variable-editor/src/components/variables/dialog/OverwriteDialog.tsx

```diff
@@ -51,29 +52,27 @@ export const OverwriteDialog = ({ table }: OverwriteProps) => {
   );
 };
 
-const addVariable = (variables: Array<Variable>, node: ProjectVarNode): AddNodeReturnType<Variable> => {
-  const lastDot = node.key.lastIndexOf('.');
-  const namespace = node.key.substring(0, lastDot);
-  let metadataType: MetadataType = '';
-  if (isMetadataType(node.type)) {
-    metadataType = node.type;
+const addVariable = (variables: Array<Variable>, node: KnownVariables): AddNodeReturnType<Variable> => {
+  let metadata: Metadata = { type: '' };
+  const nodeMetaData = node.metaData as Metadata;

```

Comment:
why is this cast needed?

Reply:
Because the function `isMetadata` (next line) takes an argument of type `Metadata`. Thinking about it, this seems indeed strange. Do you think it's better to change `isMetadata` to take an argument of type `unknown` instead? Something like:
```typescript
export const isMetadata = (metadata: unknown): metadata is Metadata => {
  return (
    typeof metadata === 'object' &&
    metadata !== null &&
    'type' in metadata &&
    isMetadataType((metadata as Metadata).type)
  );
};
```

Reply:
Yes definilty. 
The point is that isMetadata is a typeguard right? So it checks if the thing which is given is respecting the wished type. 
So the argument for it should be of type unknown and the return type is guarded as (in this case) Metadata